File: src/基础算法和其他/模拟/P11004_蓝桥杯2024省Python_B_魔法巡游/Main.ts
function check(num: number): boolean {
    if (num === 0) return true;
    let n = num;
    while (n > 0) {
        const digit = n % 10;
        if (digit === 0 || digit === 2 || digit === 4) return true;
        n = Math.floor(n / 10);
    }
    return false;
}

function main(): void {
    // @ts-ignore
    const fs = require('fs');
    const input = fs.readFileSync(0, 'utf-8').split(/\s+/);
    let ptr = 0;
    const n = parseInt(input[ptr++]);

    // @ts-ignore
    const si: number[] = new Array(n + 1).fill(0);
    // @ts-ignore
    const ti: number[] = new Array(n + 1).fill(0);

    for (let i = 1; i <= n; i++) si[i] = parseInt(input[ptr++]);
    for (let i = 1; i <= n; i++) ti[i] = parseInt(input[ptr++]);

    let inp = 1, count = 0;
    for (let i = 1; i <= n; i++) {
        const temp = (inp % 2 === 1) ? si[i] : ti[i];
        if (check(temp)) {
            count++;
            inp++;
        }
    }
    console.log(count);
}

main();

File: src/基础算法和其他/模拟/P11004_蓝桥杯2024省Python_B_魔法巡游/Main.py
def main():
    import sys
    input = sys.stdin.read().split()
    ptr = 0
    n = int(input[ptr])
    ptr += 1

    si = [0] * (n + 1)
    ti = [0] * (n + 1)

    for i in range(1, n+1):
        si[i] = int(input[ptr])
        ptr += 1

    for i in range(1, n+1):
        ti[i] = int(input[ptr])
        ptr += 1

    inp = 1
    count = 0

    for i in range(1, n+1):
        flag = False
        if inp % 2 == 1:
            temp = si[i]
        else:
            temp = ti[i]

        if temp == 0:
            flag = True
        else:
            num = temp
            while num > 0:
                digit = num % 10
                if digit in (0, 2, 4):
                    flag = True
                    break
                num = num // 10

        if flag:
            count += 1
            inp += 1

    print(count)

if __name__ == "__main__":
    main()